include/pistache/http.h

```diff
@@ -494,6 +498,15 @@ namespace Pistache
             //  automatically set to the requested encoding, if supported...
             void setCompression(const Pistache::Http::Header::Encoding _contentEncoding);
 
+#ifdef PISTACHE_USE_CONTENT_ENCODING_BROTLI
+            // Set the compression level for Brotli algorithm. Defaults to
+            //  BROTLI_DEFAULT_QUALITY...
+            void setCompressionBrotliLevel(const int _contentEncodingBrotliLevel)
```

Comment:
Before adding new functions which are conditionally compiled with `#ifdef`s, I'd like to see if there's a better way to support custom compression levels. For example, if I understand correctly, a user cannot enable both Brotli and Deflate compression for the same request, so having two `contentEncoding*Level_` variables for request surely can be avoided, right?

Reply:
As an aside, having an underscore before the parameter name seems odd to me, and if I recall correctly we don't do it anywhere else, so `_contentEncodingBrotliLevel` should be `contentEncodingBrotliLevel`

Reply:
Hey @Tachi107. The underscore isn't really my preference either. I was just trying to emulate the existing convention that appeared to be in effect.

Yes, you're correct the user can't enable multiple compression schemes, but can only request them with the intention being only one of them will be used (e.g. via q-values).

I agree that there should be a generic way to support compression levels across different codecs. I think one proposal is to have a normalized scale of [0.0, 1.0] where 0 is lowest and 1 is maximum compression.